InWorldz/InWorldz.RemoteAdmin/RemoteAdmin.cs

```diff
@@ -189,14 +195,27 @@ private void sessionTimer_Elapsed(object sender, ElapsedEventArgs e)
             }
         }
 
+        private string CreateRemoteAccessHash(string passcode)
+        {
+            return Util.CreateSaltedPasscodeHash(m_remoteAccessSalt, passcode);
+        }
+
+        private bool CheckRemoteAccessCode(string username, string password)
+        {
+            StringComparer comparer = StringComparer.OrdinalIgnoreCase;
+
+            // If a remote passcode was not set up then check with the system - however that only works on Windows under .NET.
+            return m_remoteAccessHash.Length > 0 ? 0 == comparer.Compare(CreateRemoteAccessHash(password), m_remoteAccessHash) : Util.AuthenticateAsSystemUser(username, password);
```

Comment:
This line does the backwards compat with the existing auth system that Maestro et al use.  If the remote access has is not set or is blank, the old Util method gets used.